connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteThreadRunner.scala

```diff
@@ -115,7 +115,8 @@ private[connect] class ExecuteThreadRunner(executeHolder: ExecuteHolder) extends
         case e: Throwable =>
           logDebug(s"Exception in execute: $e")
           // Always cancel all remaining execution after error.
-          executeHolder.sessionHolder.session.sparkContext.cancelJobsWithTag(executeHolder.jobTag)
+          executeHolder.sessionHolder.session.sparkContext.cancelJobsWithTag(executeHolder.jobTag,
+            "Exception in execute: " + Utils.exceptionString(e))
```

Comment:
I'm a bit worried about putting the full exception string in the reason field, as it can be very long. How about `A job with the same tag '$tagName' has failed`